### DISCUSSION TITLE
Multiselect support

### DISCUSSION THREAD

**Discussion by SaavanNanavati** [ariakit/ariakit#1929] [Questions] [answered]
Hi,

I was considering using your Combobox component because I'm trying to push my team to write more accessible code. I was wondering if the Combobox component supports multiselecting items or not. I saw you also have a Checkbox component in aria-kit, so maybe this is something I would need to implement myself by combining both? Although I'm not sure what the implications are for aria-kit for screen readers if items are multi-selected.

Also I saw .value is set as string type, can this also be string[] type for multiselect?

**Comment by diegohaz** [comment, marked as answer]
There's a work-in-progress example for a multi-selectable combobox that combines Combobox and Select (which has multi-selectable functionalities): https://ariakit.org/examples/combobox-multiple

You can reuse the `combobox-multiple.tsx` file and adjust it as you need.

**Reply by SaavanNanavati** [reply]
Thank you! But it seems like selecting multiple values doesn't actually update the input box? Is that expected?

**Reply by diegohaz** [reply]
Yes! You can render buttons using the `values` state though: https://codesandbox.io/s/ariakit-multi-select-combobox-with-tags-yuem9x

**Reply by SaavanNanavati** [reply]
Awesome!

**Reply by SaavanNanavati** [reply]
Question: when will ComboboxMultiple be fully released? It is quite awesome but we're worried that the final version could have breaking changes from this version.

**Reply by diegohaz** [reply]
That's just an example. If we change something there, that won't affect your code. I said it was a work-in-progress because it's still lacking some adjustments like adding `aria-multiselectable` to `ComboboxList` (https://github.com/ariakit/ariakit/issues/1647#issuecomment-1190581817).

The library will have breaking changes until the final v2 release though. In this case, you can stick with the current version until you have the time/need to upgrade.

**Reply by SaavanNanavati** [reply]
Ah okay got it, thanks! Also for some reason, I'm not able to type in the input when running this code. Any idea what might be causing the problem? Everything else works. It's just everytime I type in the input, nothing shows up.

**Reply by diegohaz** [reply]
I don't see any problem in Chrome/Safari/Firefox on macOS. If you can submit an issue with reproducible steps, I'll take a look.

**Reply by SaavanNanavati** [reply]
Ah actually, it looks like it was caused by taking the IDE's suggestion of updating the useEffect dependencies. So apparently I am not supposed to add them in.

**Reply by SaavanNanavati** [reply]
Is it possible to debounce the input with this example with something like react-debounce-input? Not sure if I have access to the input.
